exploration_recc_complaints/processing_recc_data.py

```diff
@@ -0,0 +1,207 @@
+
+
+def camel_case_columns(df: pd.DataFrame):
+    """
+    Transforms column names in pandas dataframe into lower and camel case.
+    """
+    cols = df.columns
+    cols = [c.lower() for c in cols]
+    cols = [c.replace(" ", "_") for c in cols]
+    df.columns = cols
+
+
+def deal_with_fit(text: str) -> str:
+    """
+    Replaces 'FiT' by "feed in tariff.
+    We do this separately from merges_expression_variations() as we do not
+    want to replace the word 'fit' by 'feed in tariff'. 
+    """
+    return text.replace("FiT", "feed in tariff")
+
+
+def merges_expression_variations(text: str) -> str:
+    """
+    Merges together different variations of the same expression.
+    E.g.: Replaces "air source heat pump" by "ashp" in text.
+    """
+    for expression in variants_same_expression.keys():
+        text = text.replace(expression, variants_same_expression[expression])
+    return text
+
+
+def process_complaint_summary(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Processes complaint summary variable by:
+    - Dealing with "FiT" expression;
+    - Creating new variable with lower case complaint text;
+    - Merging together different air source heat pump expressions;
+    - Creating a new variable with number of characters in complaint;
+    - Creating a variable with the complaint summary tokens.
+
+    Args:
+        data: dataframe with RECC data
+    Returns:
+        The processed dataframe.
+    """
+
+    data["processed_complaint_summary"] = data["complaint_summary"].apply(deal_with_fit)
+
+    data["processed_complaint_summary"] = data[
+        "processed_complaint_summary"
+    ].str.lower()
+
+    data["processed_complaint_summary"] = data["processed_complaint_summary"].apply(
+        merges_expression_variations
+    )
+
+    data["complaint_length"] = data["processed_complaint_summary"].str.len()
+
+    # there are other ways we could do this
+    data["tokens"] = data["processed_complaint_summary"].apply(
+        lambda x: re.sub("[^A-Za-z0-9]+", " ", x)
+    )
+
+    data["tokens"] = data["tokens"].apply(lambda x: word_tokenize(x))
+
+    data["stems"] = data["tokens"].apply(stemming)
+
+    data["lemmas"] = data["tokens"].apply(lemmatising)
+
+    return data
+
+
+def create_dummy_variables_and_total(
+    data: pd.DataFrame, variable: str, prefix: str
+) -> pd.DataFrame:
+    """
+    Creates dummy variables based on an existing variable in the data frame.
+    Also creates a variable with the totals (number of dummy variables with 1 for each instance).
+
+    Args:
+        data: dataframe with RECC data
+        variable: variable to split into dummy variables
+        prefix: prefix to add to each dummy variable
+    Returns:
+        The processed dataframe.
+
+    """
+
+    unique_values = list(set(data[variable].unique().sum().split("; ")[:-1]))
+
+    dummy_variable_names = [prefix + ":" + c for c in unique_values]
+
+    for c in range(len(unique_values)):
```

Comment:
will do, to avoid confusion :) although `c` doesn't exist outside `dummy_variable_names`, so wouldn't be a problem to use it again (but I see that we are using it do define two different things)